include/ipc/ipc_service.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_IPC_SERVICE_IPC_SERVICE_H_
+#define ZEPHYR_INCLUDE_IPC_SERVICE_IPC_SERVICE_H_
+
+#include <stdio.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief IPC Service API
+ * @defgroup ipc_service_api IPC service APIs
+ * @{
+ */
+
+/** @brief Event callback structure
+ *  It is registered during endpoint registration.
+ *  This structure is packed into the endpoint configuration.
+ */
+struct ipc_service_cb {
+	/** @brief Bind was successful
+	 *  @param priv Private user data.
+	 */
+	int (*bound)(void *priv);
+
+	/** @brief New packet arrived.
+	 *  @param data Pointer to data buffer.
+	 *  @param len Length of @a data.
+	 *  @param priv Private user data.
+	 */
+	int (*received)(void *data, size_t len, void *priv);
+
+	/** @brief An error occurred
+	 *  @param message Error message.
+	 *  @param priv Private user data.
+	 */
+	int (*error)(const char *message, void *priv);
+};
+
+/** @brief Endpoint instance
+ *  Content is not important for user of the API.
+ *  It is implemented in a specific backend.
+ */
+struct ipc_ept;
+
+/** @brief Endpoint configuration structure */
+struct ipc_ept_cfg {
+	const char *name;           /**< @brief Name of the endpoint */
+	struct ipc_service_cb cb;   /**< @brief Event callback structure */
+	int prio;                   /**< @brief Endpoint priority */
+							    /* If the backend supports priorities*/
+	void *priv;                 /**< @brief Private user data */
+};
+
+/** @brief Register IPC endpoint
+ *
+ *  Registers IPC endpoint to enable communication with a remote device.
+ *
+ *  The same function registers endpoints for both master and slave devices.
+ *
+ *  @param ept Endpoint object.
+ *  @param cfg Endpoint configuration.
```

Comment:
Please clarify which module is responsible to keep buffers for `name` field.

Is value of `name` copied by this function and caller can free the memory buffer containing this value after this function finishes? Or is the caller responsible to maintain static buffer containing `name` value?

It should be explicitly stated to make sure all backend implementations follow the same rule.

Reply:
The caller is responsible for allocation and maintenance of memory for `name`. 
The backend implementation only works with a pointer to `name`.
It will be necessary to add information about this.